File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;
import java.util.stream.Collectors; 


class Solution {
    public int solution(int[] scoville, int K) {
        PriorityQueue<Integer> heap = new PriorityQueue(
            Arrays.stream(scoville).boxed().collect(Collectors.toList())
        );
        
        int result = 0;
        while (2 <= heap.size() && heap.peek() < K) {
            int a = heap.poll();
            int b = heap.poll();
            
            heap.add(a + b * 2);
            result++;
        }
        
        if (heap.peek() < K)
            return -1;
        return result;
    }
}

File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
from heapq import heappush
from heapq import heappop
from heapq import heapify

def solution(scoville, K):
    heap = []
    for num in scoville:
        heappush(heap, num)
    
    answer = 0
    while heap:
        if len(heap) < 2:
            break
            
        if K <= heap[0]:
            break
            
        a, b = heappop(heap), heappop(heap) 
        heappush(heap, a + b * 2)
        answer += 1
        
    return answer if K <= heap[0] else -1
    